### PR TITLE
ci: use LTS node version in workflows

### PR DESCRIPTION
**Pull request by snitin315**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[x] Other, please explain: CI

#### What changes did you make? (Give an overview)

Use the [Node.js LTS version syntax](https://github.com/actions/setup-node#supported-version-syntax) so we don't have to manually update the node version in the future.

#### Is there anything you'd like reviewers to focus on?

No

### CODE REVIEW

**Review by bmish — APPROVED** [review]
Is it possible to use this under the `test_on_node` job too?

**Comment by snitin315** [comment]
@bmish I feel it should be fine for the `test_on_node` job as we are testing against all major node versions ESLint supports there.

**Comment by bmish** [comment]
but why can't this use LTS?

```
        include:
        - os: windows-latest
          node: "16.x"
        - os: macOS-latest
          node: "16.x"
```

**Comment by snitin315** [comment]
Oh! I missed those instances, I'll update.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -65,7 +65,7 @@ jobs:
     - uses: actions/checkout@v3
     - uses: actions/setup-node@v3
       with:
-        node-version: '16'
+        node-version: 'lts/*'
```
Comment (mdjermanovic) [inline .github/workflows/ci.yml:68]: I think we'd like the Browser Test environment to be in sync with Netlify, where the builds are still on Node 16.

Comment (snitin315) [reply .github/workflows/ci.yml:68]: That makes sense, I'll revert this change.

Comment (snitin315) [inline .github/workflows/ci.yml:68]:
```suggestion
        node-version: '16'
```

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks!

File: .github/workflows/ci.yml (hunk 1 above)
Comment (bmish) [reply .github/workflows/ci.yml]: There should be a comment on this line explaining that.

**Pull request merged**